src/main/java/org/folio/dew/repository/FTPObjectStorageRepository.java

```diff
@@ -0,0 +1,136 @@
+package org.folio.dew.repository;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.net.ftp.FTP;
+import org.apache.commons.net.ftp.FTPClient;
+import org.apache.commons.net.ftp.FTPReply;
+import org.folio.dew.config.properties.FTPProperties;
+import org.folio.dew.exceptions.FtpException;
+import org.springframework.beans.factory.ObjectFactory;
+import org.springframework.stereotype.Repository;
+
+import lombok.extern.log4j.Log4j2;
+
+@Log4j2
+@Repository
+public class FTPObjectStorageRepository {
+
+  private FTPClient ftpClient;
+
+  private final ObjectFactory<FTPClient> ftpClientFactory;
+  private final FTPProperties ftpProperties;
+
+  public FTPObjectStorageRepository(ObjectFactory<FTPClient> ftpClientFactory, FTPProperties ftpProperties) {
+    this.ftpProperties = ftpProperties;
+    this.ftpClientFactory = ftpClientFactory;
+  }
+
+  public boolean login(String ftpUrl, String username, String password) throws URISyntaxException {
+    ftpClient = ftpClientFactory.getObject();
+    if (!isUriValid(ftpUrl)) {
+      throw new URISyntaxException(ftpUrl, "URI should be valid ftp path");
+    }
+    boolean isLogin = false;
+    try {
+      URI url = new URI(ftpUrl);
+      String server = url.getHost();
+      int port = url.getPort() > 0 ? url.getPort() : ftpProperties.getDefaultPort();
+      ftpClient.connect(server, port);
+      log.info("Connected to {}:{}", server, port);
+      int reply = ftpClient.getReplyCode();
+      if (!FTPReply.isPositiveCompletion(reply)) {
+        throw new FtpException(ftpClient.getReplyCode(), ftpClient.getReplyString().trim());
+      }
+
+      if (ftpClient.login(username, password)) {
```

Comment:
I can propose refactor this part of code - extract it to separate method to simplify code.